pkg/kubelet/certificate/kubelet.go

```diff
@@ -299,9 +301,8 @@ func (m *kubeletServerCertificateDynamicFileManager) Current() *tls.Certificate
 
 // Start starts watching the certificate and key files
 func (m *kubeletServerCertificateDynamicFileManager) Start() {
-	var ctx context.Context
-	ctx, m.cancelFn = context.WithCancel(context.Background())
-	go m.dynamicCertificateContent.Run(ctx, 1)
+	m.ctx, m.cancelFn = context.WithCancel(context.Background())
+	go m.dynamicCertificateContent.Run(m.ctx, 1)
```

Comment:
didn't dig too much into the code , but it seems this method has to be rewritten to pass the context through Start(ctx)